chain/evm/provider/transactor_generator.go

```diff
@@ -0,0 +1,73 @@
+package provider
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// TransactorGenerator is an interface for generating bind transactor instances
+type TransactorGenerator interface {
+	// Generate creates a new Solana keypair and returns the private key
+	Generate(chainID *big.Int) (*bind.TransactOpts, error)
+}
+
+var (
+	_ TransactorGenerator = (*transactorFromRaw)(nil)
+	_ TransactorGenerator = (*transactorRandom)(nil)
+)
+
+// TransactorFromRaw creates a new instance of the privateKeyFromRaw generator with the raw private
```

Comment:
```suggestion
// TransactorFromRaw creates a new instance of the transactorFromRaw generator with the raw private
```
However i am not sure we should mention `transactorFromRaw` in the go doc since `transactorFromRaw` is private to only this repo and consumers do not need to know about it.